.github/workflows/release.yml

```diff
@@ -0,0 +1,59 @@
+name: Release
+
+on:
+  push:
+    tags:
+    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      packages: write
+      contents: write
+
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: on
+      COSIGN_EXPERIMENTAL: "true"
+
+    steps:
+    - uses: actions/setup-go@f6164bd8c8acb4a71fb2791a8b6c4024ff038dab # v2.2.0
```

Comment:
Did you generate these in some way, or is it literally just picking the right sha1 for immutability?

Reply:
get the sha from the release and pin by sha for immutability :) it is a good practice